New package: SciTypes v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: SciTypes
- Repository: https://github.com/JuliaML/SciTypes.jl
- Created by: @eliascarv
- Version: v0.1.0
- Commit: 7e9403ff094a1c8ea2cf97c17f02fb0bb6ac53a7
- Reviewed by: @eliascarv
- Reference: https://github.com/JuliaML/SciTypes.jl/commit/7e9403ff094a1c8ea2cf97c17f02fb0bb6ac53a7#commitcomment-129640205
- Description: Scientific types for data science
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd610e536691739724d995d5af37e117ef0b422b90f1220d5ab9a977741182275c7292cd6f7c6eb7570082609accb6d76d7ae6c1259c15fd38eee3190e37e69ac1aed1884e3af8284f318d5c97080163b95fce4360003f16c1fb1dbe5fa4b1e541e364f95c7525de0bd2287517ee93c02572d0b2b6c29b1397dce95d560d6ad2ad4dbead3931941a4ad1211332ef01676c182c81ba7295af54d0e158080166d8772f50c1b033f28d42e1ba5b1420bf05ce8 -->

goerz commented:
Please make sure to add some documentation before registering, at least a basic usage example in the README

juliohm commented:
@goerz thank you for the feedback. Example added to the README.

juliohm commented:
Regarding the similarity with SumTypes.jl and SubTypes.jl, it shouldn't be an issue. These packages have very different purpose.

JuliaRegistrator committed:
New package: SciTypes v0.1.0

UUID: 6cb2f572-2d2b-4ba6-bdb3-e710fa044d6c
Repo: https://github.com/JuliaML/SciTypes.jl.git
Tree: fb49e8a042f2af22b470cdb4057cfa3580993f62

Registrator tree SHA: f73a20c99934db92a256057d0d83ba394036a701

ChrisRackauckas commented:
> with a more sensible set of defaults for data science

DataSciTypes?

juliohm commented:
I think SciTypes.jl is fine and shorter to type. Thanks for the suggestion
though.

Em seg., 16 de out. de 2023 07:56, Christopher Rackauckas <
***@***.***> escreveu:

> with a more sensible set of defaults for data science
>
> DataSciTypes?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaRegistries/General/pull/93099#issuecomment-1764220427>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZQW3OGWTNCMZVOS3ZHLOLX7UHG3AVCNFSM6AAAAAA5ZCOU2WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRUGIZDANBSG4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

ChrisRackauckas commented:
It's just very non-descriptive. The package name and the abstract type has nothing to do with the actual contents, so it's quite confusing. `scitype(::Type{<:Number}) = Continuous` `scitype(::Type{<:Integer}) = Categorical`, Integers are categorical yes, but is that... science? NumberTypes or better yet, NumberTraits or something would be a more descriptive name. I wouldn't block this but given that the naming is so odd, I wouldn't choose to merge it myself.

juliohm commented:
The traits apply to non-number types as well so NumberTypes or NumberTraits
are not good names. Scientific types is the name adopted in the ecosystem
already, and we are simply reusing it.

Em seg., 16 de out. de 2023 08:02, Christopher Rackauckas <
***@***.***> escreveu:

> It's just very non-descriptive. The package name and the abstract type has
> nothing to do with the actual contents, so it's quite confusing. scitype(::Type{<:Number})
> = Continuous scitype(::Type{<:Integer}) = Categorical, Integers are
> categorical yes, but is that... science? NumberTypes or better yet,
> NumberTraits or something would be a more descriptive name. I wouldn't
> block this but given that the naming is so odd, I wouldn't choose to merge
> it myself.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaRegistries/General/pull/93099#issuecomment-1764228726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZQW3M6P47LXUWMR7XPDZ3X7UH25AVCNFSM6AAAAAA5ZCOU2WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONRUGIZDQNZSGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

ViralBShah commented:
Reading through this, I found ScientificTypes itself not easy to understand as a name. The issue here is that now that I read about ScientificTypes, SciTypes would create confusion - as in which one came first, and which one is the lightweight version.

What do you think about LightScientificTypes.jl, if that describes the intention of the package? Alternatively, how about DataScienceTypes.jl, if based on the README, this is a package with better defaults for DataScience?

juliohm commented:
We will go with DataScienceTypes.jl, thank you for the suggestion.
